woocommerce/wc-admin
# Add performance tracks to onboarding flow

Fixes #7919 

This adds one extra onboarding track for the business details free extensions step to add info about installation time.
Also adding one extra prop for the product types step as we sometime install WC Pay there.

### Screenshots

<img width="1050" alt="Screen Shot 2021-12-16 at 4 31 30 PM" src="https://user-images.githubusercontent.com/2240960/146444895-c43568e8-6805-49cc-8b10-ac28a98b8f0f.png">

cc @pmcpinto & @becdetat I added a new track `storeprofiler_store_business_features_installed_and_activated` for the install time props, see the example above. Does this look ok, or would you prefer to tag it on to one of the existing tracks. I decided on a new track mostly to keep thinks semi clean and allowing for easy removal if we decide we don't want it anymore.

### Detailed test instructions:

- Start a fresh site and enable the console log for tracks -> `localStorage.setItem( 'debug', 'wc-admin:*' );`
- Walk through the onboarding flow and make sure you select all or some of the free extensions
- Once that step is finished it should show a new track called -> `storeprofiler_store_business_features_installed_and_activated`
- This should include the install time for each selected plugin, install time is put in several time slots 0-2sec, 3-5sec, etc...
- Now delete the installed plugins and enable [this plugin](https://gist.github.com/louwie17/c7f9a023e05479919ed809564a656869) (which causes Jetpack or Google Listings and Ads to fail)
- Go through the onboarding again, making sure to have Jetpack or Google Listings and Ads selected
- Notice how the step fails now and the track is still triggered, this time with `success` set to false and a `failed_extensions` list.
